meetings/working-groups/collection-literals/Core-interface-target-type-proposal.md

```diff
@@ -0,0 +1,248 @@
+
+Indeed, if it were not for `params T[]` (a full 50% of all collection-taking methods), `IEnumerable<T>` would be the most commonly taken collection by far for the ecosystem.
+
+Ideally, we would ship with support for everything, but we've currently made judicious moves from C#12 to C#13 based on complexity, but also based on impact.  For example, `Dictionary expressions` were moved to C#13 to lighten our load, and because data indicates that APIs that consume those dictionary types are only <3% of all apis that take collections in the first place.  `Natural type` support has also been pushed out because the complexity is felt to be substantive enough to warrant more time.
+
+With how important these interface types are though, we do not believe pushing out from C# 12 will allow us to ship a viable and coherent story to customers.  
+
+### What factors are at play?
+
+In any interesting design, there are many factors that must be assessed and considered as a holistic whole.  For collection expressions, the broad areas we want the land the feature are include, but are not limited to:
+
+1. Simplicity. Ideally the feature works in a fashion that is both simple to explain and simple for users to understand. Using some of our modern terminology, we'd like to avoid 'decoder rings' when people use it.
+
+2. Universality.  This restates the background context.  Namely that we want to meet the literal 97% case at launch.  This means needing good stories for arrays, spans, BCL concrete collections *and* BCL core interfaces.
+
+3. Brevity.  It is a strong goal of this feature that users be able to just write the simple, idiomatic, collection expression form without the need to do things like add coercive casts to commonly appease the compiler.  Specifically, for casts, once you add them (e.g. `(List<int>)[1, 2, 3]`) then the benefit of the feature as a whole is vastly diminished. Where the new form isn't substantively better than the existing form (just 2 characters saved over `new List<int> {1, 2, 3}`).  Unlike other features, the cliff here is very steep, often fundamentally negating the idea that this is a valuable feature in the first place.  Many parts of the design (especially broad adoption of target-typing) has been entirely around ensuring users can just write the simple expression form and almost never have to do things to appease the language.
+
+4. Performance. A core pillar of collection expressions that we are both evangelizing it by, and which we are seeing customers resonate with, is the idea of: "Absent external information unavailable to the compiler, the compiler should almost always do as well or better than the user could.  Sometimes much more so."  Because the user can write a simple `[a, b, c, d, e]` expression, without having to explicitly state what is going on, and because we can provide so much smart understanding to each situation, we can heavily optimize.  For example:
```

Comment:
```suggestion
4. Performance. A core pillar of collection expressions that we are both evangelizing it by, and which we are seeing customers resonate with, is the idea of: "Absent external information unavailable to the compiler, the compiler should almost always do as well or better than the user could.  Often much more so.  And almost certainly with clearer code for the user to write."  Because the user can write a simple `[a, b, c, d, e]` expression, without having to explicitly state what is going on, and because we can provide so much smart understanding to each situation, we can heavily optimize.  For example:
```